roles/docker_storage_verify/tasks/main.yml

```diff
@@ -0,0 +1,42 @@
+---
+# vim: set ft=ansible:
+#
+- name: Include distribution vars
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_distribution|lower }}-{{ ansible_distribution_major_version }}.yml"
+    - "{{ ansible_distribution|lower }}.yml"
+
+- name: Fail if required variables are not defined
+  fail:
+    msg: "The required variables are not defined.  Check files under roles/docker_storage_verify/vars"
+  when: driver is undefined
+
+  # Maximum efficiency!  The output from 'docker info' contains a line:
+  #
+  #   Storage Driver: <value>
+  #
+  # So we grab the value, split on the colon + space, and compare the value
+  # to the expected driver.  All in one task!
+- name: Check Docker storage driver
+  shell: docker info | grep '^Storage Driver'
+  register: storage_driver
+  failed_when: "'{{ storage_driver.stdout.split(': ')[1] }}' != '{{ driver }}'"
```

Comment:
i'm that guy only using the latest and greatest of everything and expecting everyone else to just update :) - sorry about that - maybe add a comment to the code to mention updating it when it can be updated